### PR TITLE
packet.h clarification

### PR DESCRIPTION
**Pull request by sader1992**
* **Server Mode**: both

* **Description of Pull Request**: 

i see a lot of people edit this without known , so this line just to tell them do not edit it .

### CODE REVIEW

**Comment by lighta** [comment]
hmm, the comment could be misleading if you want to lower the packet_version.
What is the common problematic ?
Perhaps if we want to details we should put a note to see doc/packet_config.txt or something of that sort ?

**Comment by sader1992** [comment]
@lighta 
when people edit the packets.h to change the `PACKETVER` to 2017 clients RagexeRE
they see like this
`#ifndef PACKETVER
	#define PACKETVER 20151104
#endif

#ifndef PACKETVER_RE
	/// From this point on only kRO RE clients are supported
	#if PACKETVER > 20151104
		#define PACKETVER_RE
	#endif
#endif`
they have to edit `#define PACKETVER 20151104`
but they also edit `#if PACKETVER > 20151104`
and this will lead for this error
screenrathena014
so i would move this option from here to prevent misleading when they change the `PACKETVER`
but since my knowledge in C/C++ is limited i did make a statement to those who want to change it
i did think if i made the statement like
"Don't edit if you use RagexeRE"
but then they will wander what they have to do if they use Ragexe

**Comment by Daegaladh** [comment]
I think you should add a comment in #define PACKETVER too, something like:

> 
> Do not edit this line. To set your client version, please do this instead.
> In Linux: run the following command: ./configure --enable-packetver=YYYYMMDD
> In Windows: add this line in your src\custom\defines_pre.h file: #define PACKETVER YYYYMMDD

**Comment by lighta** [comment]
@sader1992 so basically, the comment should be on the #if with condition.
(i.e #if PACKETVER > 20151104). here only. The 1st part is perfectly valid to edit is just that they getting confuse and chose extra condition.

Now we could put your comment warning but there a little issue with that, is that we have tons of macro like this, so we may end up doing tons of comment.
To be fair editing a condition is never a good idea if you don't know why it was there...

Now we could hide that PACKETVER_RE enabling precondition. Since one of the issue is that this is in config/ area so people are allowed to edit stuff here.
What we could do is moved down those PACKETVER minimun version a little down to the file after the 
/**
* No settings past this point
**/

File: src/config/packets.h [hunk 1]
```diff
@@ -9,16 +9,12 @@
  **/
 
 #ifndef PACKETVER
+	///Do not edit this line. To set your client version, please do this instead.
+	///In Linux: run the following command: ./configure --enable-packetver=YYYYMMDD
+	///In Windows: add this line in your src\custom\defines_pre.h file: #define PACKETVER YYYYMMDD
```
Comment (aleos89) [inline src/config/packets.h]: The `defines_pre.h` can also be used for Linux, not just Windows.

The documentation is good, we just don't want it to be misleading that it's only applicable to Windows builds.

File: src/config/packets.h [hunk 2]
```diff
@@ -9,16 +9,12 @@
...
 	#define PACKETVER 20151104
 #endif
 
-#ifndef PACKETVER_RE
-	/// From this point on only kRO RE clients are supported
-	#if PACKETVER > 20151104
-		#define PACKETVER_RE
-	#endif
-#endif
-
```
Comment (aleos89) [inline src/config/packets.h]: Is there a reason this was moved down?

Comment (sader1992) [reply src/config/packets.h]: some people get confused by that `#define PACKETVER 20151104` is the same of `#if PACKETVER > 20151104` so they edit both
while we can't control the server owner option to edit the version in here when he edit both to put his version while he use RagexeRE +2016 he will get the error in the screenshot above
moving it down will decrease the error rate by the servers owner
however if people start to use the other methods to set there version , no one would get the error and get confused BY `#define PACKETVER` AND `#if PACKETVER`

File: src/config/packets.h (hunk 1 above)
Comment (Daegaladh) [reply src/config/packets.h]: What about...

> ///In Windows: add this line in your src\custom\defines_pre.h file: #define PACKETVER YYYYMMDD
> ///In Linux: The same as above, or run the following command: ./configure --enable-packetver=YYYYMMDD

**Comment by sader1992** [comment]
after being a while
i do believe it's better to Encourage people to edit the version in the other files rather then changing the check place

**Pull request merged**